In [1]:
import numpy as np
import pandas as pd
import os
print(os.listdir("../input"))

['resources.csv', 'train.csv', 'test.csv', 'sample_submission.csv']


In [2]:
data = pd.read_csv('../input/train.csv')

In [3]:
grades = data[['project_grade_category','project_is_approved']]
grades_approved = grades[grades['project_is_approved'] == 1].drop('project_is_approved',axis=1)
grades_rejected = grades[grades['project_is_approved'] == 0].drop('project_is_approved',axis=1)

In [4]:
grades_approved_count = grades_approved.groupby('project_grade_category').size()
grades_rejected_count = grades_rejected.groupby('project_grade_category').size()

In [5]:
grades_approved_df = pd.DataFrame()
grades_approved_df['count'] = grades_approved_count

In [6]:
grades_rejected_df = pd.DataFrame()
grades_rejected_df['count'] = grades_rejected_count

In [7]:
grades_count = grades_approved_df.join(grades_rejected_df, how='inner',
                                          lsuffix='_approved',
                                          rsuffix='_rejected')

In [8]:
grades_count['count_total'] = grades_count['count_approved'] + grades_count['count_rejected']
grades_count['probability_approved'] = grades_count['count_approved'] / grades_count['count_total']

In [9]:
grades_count['probability_approved']

project_grade_category
Grades 3-5       0.853977
Grades 6-8       0.843459
Grades 9-12      0.835236
Grades PreK-2    0.847124
Name: probability_approved, dtype: float64

In [10]:
approved = data[data['project_is_approved'] == 1]
rejected = data[data['project_is_approved'] == 0]
num_approved = approved['id'].count()
num_rejected = rejected['id'].count()
print(num_approved/(num_approved+num_rejected))

0.8476823374340949


So as an important note, the global approval rate seems to be around 85%. I'm not sure yet how to figure out what a good interval is, but for now I will consider anything within a couple percent of that to be too average to consider as a good predictor for approval/rejection.